scheduler/src/Control/Scheduler/Internal.hs

```diff
@@ -25,6 +26,16 @@ import Control.Scheduler.Queue
 import Data.IORef
 import Data.Primitive.Array
 
+-- | Computated outcome of scheduled jobs.
+--
+-- @since 1.4.2
+data Results a
+  = Finished ![a]
+  -- ^ Finished normally with all scheduled jobs completed
+  | FinishedEarly ![a] !a
+  -- ^ Finished early by the means of `Control.Scheduler.terminate`.
```

Comment:
Ah, so this can't be `NonEmpty` because we don't know if anything else was actually scheduled. And even if it were, nothing else might have completed.

Reply:
It could be, but only if the argument that is supplied to `terminate` was added to the list, so it could be constructed with either `FinishedEarly (reverse (x :| xs))`, where `xs` is the FILO stack with computed results and `x` is the argument to `terminate`. But I thought it would be less ambiguous to keep it as a separate argument: `FinishedEarly (reverse xs) x`

Other than that, you are absolutely right:
>  because we don't know if anything else was actually scheduled. And even if it were, nothing else might have completed.

Reply:
It should be fine as `[a]` after all, I think.